app/services/email_service/ses/synchronize.rb

```diff
@@ -0,0 +1,74 @@
+module EmailService::SES::Synchronize
+
+  AddressStore = EmailService::Store::Address
+
+  BATCH_SIZE = 1000
+
+  module_function
+
+  def run_synchronization!(ses_client)
+    ses_client.list_verified_addresses.on_success { |vaddrs|
+      offset = 0
+      addresses = AddressStore.load_all(limit: BATCH_SIZE, offset: offset)
+
+      while addresses.present?
+        update_statuses(build_sync_updates(addresses, vaddrs))
+
+        offset += BATCH_SIZE
+        addresses = AddressStore.load_all(limit: BATCH_SIZE, offset: offset)
+      end
+    }
+  end
+
+  def build_sync_updates(addresses, verified_addresses)
+    vaddrs = verified_addresses.to_set
+    updates = { verified: [], expired: [], touch: [], none: [] }
+
+    addresses.each do |a|
```

Comment:
Yeah, that's cleaner put contained mutability chosen on purpose as an easy optimization.